packages/core/saved-objects/core-saved-objects-base-server-internal/src/validation/validator.ts

```diff
@@ -22,36 +23,61 @@ import { createSavedObjectSanitizedDocSchema } from './schema';
 export class SavedObjectsTypeValidator {
   private readonly log: Logger;
   private readonly type: string;
+  private readonly defaultVersion: string;
   private readonly validationMap: SavedObjectsValidationMap;
+  private readonly orderedVersions: string[];
 
   constructor({
     logger,
     type,
     validationMap,
+    defaultVersion,
   }: {
     logger: Logger;
     type: string;
     validationMap: SavedObjectsValidationMap | (() => SavedObjectsValidationMap);
+    defaultVersion: string;
   }) {
     this.log = logger;
     this.type = type;
+    this.defaultVersion = defaultVersion;
     this.validationMap = typeof validationMap === 'function' ? validationMap() : validationMap;
+    this.orderedVersions = Object.keys(this.validationMap).sort(Semver.compare);
   }
 
-  public validate(objectVersion: string, data: SavedObjectSanitizedDoc): void {
-    const validationRule = this.validationMap[objectVersion];
-    if (!validationRule) {
-      return; // no matching validation rule could be found; proceed without validating
+  public validate(document: SavedObjectSanitizedDoc, version?: string): void {
+    const docVersion =
+      version ??
+      document.typeMigrationVersion ??
+      document.migrationVersion?.[document.type] ??
+      this.defaultVersion;
```

Comment:
So I started by removing the `version` parameter altogether, assuming that retrieving the version from the document's `typeMigrationVersion` or `migrationVersion` was going to be sufficient.

Then I realized... some teams defined schemas for types that did not register migrations for ages. For example, the latest migration for `space` is `6.6.0`, but it has a schema defined for `8.8.0`. And then I understood why the `version` parameter was here in the first place and added it back.

I kept the logic retrieving the version from the document though, as I suspect it may be useful later for zdt and version cohabitation.